tests/requirements.txt

```diff
@@ -1,5 +1,22 @@
-pyright
```

Comment:
> Removed pyright from tests/requirements.txt as it's not needed by CLI/CI
> Since pip installing pyright also installs node, this also gives the choice to maintainers of how they wanna install pyright

Do we want to give people this choice? For example, that also gives them the choice of not being running the latest pyright, which we probably don't want.

Actually, we may want to continue using `pyright-python` and set [`PYRIGHT_PYTHON_PYLANCE_VERSION=latest-prerelease`](https://github.com/RobertCraigie/pyright-python?tab=readme-ov-file#keeping-pyright-and-pylance-in-sync) to keep pyright in sync with Pylance's copy of pyright.

Reply:
I think [PYRIGHT_PYTHON_PYLANCE_VERSION](https://github.com/RobertCraigie/pyright-python?tab=readme-ov-file#keeping-pyright-and-pylance-in-sync) may be newer than this PR. I'll add that.
Anyway I forgot to change it back after https://github.com/microsoft/python-type-stubs/pull/301#discussion_r1589653460